subt/main.py

```diff
@@ -545,16 +547,17 @@ def on_acc(self, timestamp, data):
                 raise Collision()
 
     def on_artf(self, timestamp, data):
-        if self.offset is None:
+        if self.offset is None or self.orientation is None or self.xyz is None:
             # there can be observed artifact (false) on the start before the coordinate system is defined
             return
-        artifact_data, deg_100th, dist_mm = data
+        artifact_data, dx, dy, dz = data
+        vector = [dx, dy, dz]
```

Comment:
Can the data come as [`artifact_data`, [dx, dy, dz]]? So that there are 2 components - first is artifact type and second is 3d vector of relative position.

Reply:
Good point - now is the good moment to do it - the data are not backward compatible here. I used the same types as output `artf_xyz` which is the only discouraging detail (but there is list of reports, i.e. it is not 1:1).

Reply:
I change that here, but not the rest of all `artf_xyz` reports - please add it to the dept list, thanks.